web/src/containers/QuarterlyBudgetSummary.js

```diff
@@ -0,0 +1,120 @@
+import PropTypes from 'prop-types';
+import React, { Fragment } from 'react';
+import { connect } from 'react-redux';
+
+import { t } from '../i18n';
+import { formatMoney } from '../util/formats';
+
+const FUNDING_SOURCES = ['hit', 'hie', 'mmis'];
+const QUARTERS = [1, 2, 3, 4];
+const COLORS = ['teal', 'green', 'yellow'];
+const CATEGORY_LOOKUP = {
+  statePersonnel: 'Project state staff',
+  expenses: 'Non-personnel',
+  contractors: 'Contracted resources',
+  combined: 'Total Enhanced FFP'
+};
+
+const color = idx => `bg-${COLORS[idx] || 'gray'}`;
+
+const QuarterlyBudgetSummary = ({ budget }) => {
+  const { years } = budget;
+
+  // wait until budget is loaded
+  if (!years.length) return null;
+
+  return (
+    <div>
+      {FUNDING_SOURCES.map(source => {
+        const data = budget[source];
+        return (
+          <div key={source} className="mb3">
+            <h4 className="mt0">{source.toUpperCase()}</h4>
+            <div className="overflow-auto">
+              <table
+                className="h6 table-fixed table-bordered table-budget"
+                style={{ minWidth: 1200 }}
+              >
+                <thead>
+                  <tr>
+                    <th style={{ width: 160 }} />
+                    {years.map((year, i) => (
+                      <th
+                        key={year}
+                        className={`white center ${color(i)}`}
+                        colSpan="5"
+                      >
+                        {t('ffy', { year })}
+                      </th>
+                    ))}
+                    <th className="bg-black white center">
+                      {t('table.total')}
+                    </th>
+                  </tr>
+                  <tr>
+                    <th />
+                    {years.map((year, i) => (
+                      <Fragment key={year}>
+                        {QUARTERS.map(q => (
+                          <th key={q} className="right-align">
+                            {t('table.quarter', { q })}
+                          </th>
+                        ))}
+                        <th className={`right-align ${color(i)}-light`}>
+                          {t('table.subtotal')}
+                        </th>
+                      </Fragment>
+                    ))}
+                    <th className="bg-gray-light" />
+                  </tr>
+                </thead>
+                <tbody>
+                  {Object.keys(data).map(key => (
```

Comment:
Does this assume that `combined` will always be the last key on the data object?  I reckon since the budget data comes from the reducer, we can make that assertion, but if so, we should probably put some comments in the reducer noting that the property order matters.  What do you think?

Reply:
good point, i don't like this reliance -- will tweak